packages/server/src/analytics/logic.ts

```diff
@@ -27,28 +59,67 @@ export const calculateSignups = (
 
   let todaySignups = incrementDaily() ? storedTodaySignups : 0
   let totalSignups = storedTotalSignups
+  const state: CalculatedSignups['state'] = {
+    lastQueryTime: storedState.lastQueryTime,
+    values: makeAnalyticsPairForRecord(storedState.values, !incrementDaily()),
+  }
+  const { lastQueryTime: stateLastQueryTime } = state
 
-  // We process this query differently depending if this is the first time
-  // these analytics have run or not.
+  // There are two possible cases where we fetch for all entries in our DB.
   //
-  // When running for the first time, we'll have to manually iterate
-  // through the array in order to detect which records happened at queryDateTime.
+  // The first when this script is run for the first time, in this case we
+  // simply calculate the values based of whether they happened today or not.
   //
-  // Otherwise we can easily increment todaySignups/totalSignups based
-  // on the size of the snapshot.
-  if (storage.lastQueryTime === 0) {
-    snapshot.forEach(d => {
+  // The second case is more complicated, it happens when we have never queried
+  // for per-state-signups (but have already queried something before). This scenario
+  // didn't always exist, and to keep backward compatibility with the previous version
+  // of this script we query all entries again but pay attention to repeated values.
+  //
+  // To avoid incrementing repeated entries when looping from the beginning,
+  // we always check for the respective `lastQueryTime`
+  const loopingFromTheStart = !storage.lastQueryTime || !stateLastQueryTime
+  const todayMidnight = new Date(
+    queryDateTime.getFullYear(),
+    queryDateTime.getMonth(),
+    queryDateTime.getDate(),
+  )
+
+  if (loopingFromTheStart) {
+    snapshot.forEach(entry => {
+      const entryState = getImplementedState(entry.get('address.state') ?? '')
```

Comment:
Could you not do `entry?.address?.state`?

Reply:
I'm using `FirestoreService.query` to wrap the query now so it allow us to do that, should've done this in the first place